eva-accession-release/src/test/java/uk/ac/ebi/eva/accession/release/io/AccessionedVariantMongoReaderTest.java

```diff
@@ -0,0 +1,220 @@
+
+    private static final String DBSNP_CLUSTERED_VARIANT_ENTITY = "dbsnpClusteredVariantEntity";
+
+    private static final String RS_1 = "869808637";
+
+    private static final String RS_2 = "869927931";
+
+    private static final String RS_3 = "347048227";
+
+    private AccessionedVariantMongoReader reader;
+
+    private ExecutionContext executionContext;
+
+    @Autowired
+    private MongoClient mongoClient;
+
+    //Required by nosql-unit
+    @Autowired
+    private ApplicationContext applicationContext;
+
+    @Rule
+    public MongoDbRule mongoDbRule = new FixSpringMongoDbRule(
+            MongoDbConfigurationBuilder.mongoDb().databaseName(TEST_DB).build());
+
+    @Before
+    public void setUp() throws Exception {
+        executionContext = new ExecutionContext();
+        reader = new AccessionedVariantMongoReader(ASSEMBLY_ACCESSION_1, mongoClient, TEST_DB);
+    }
+
+    @Test
+    public void readTestDataMongo() {
+        MongoDatabase db = mongoClient.getDatabase(TEST_DB);
+        MongoCollection<Document> collection = db.getCollection(DBSNP_CLUSTERED_VARIANT_ENTITY);
+
+        AggregateIterable<Document> result = collection.aggregate(reader.buildAggregation())
+                                                       .allowDiskUse(true)
+                                                       .useCursor(true);
+
+        MongoCursor<Document> cursor = result.iterator();
+
+        List<Variant> variants = new ArrayList<>();
+        while (cursor.hasNext()) {
+            Document clusteredVariant = cursor.next();
+            Variant variant = reader.getVariant(clusteredVariant);
+            variants.add(variant);
+        }
+        assertEquals(2, variants.size());
+     }
+
+    @Test
+    public void reader() throws Exception {
+        List<Variant> variants = readIntoList();
+        assertEquals(2, variants.size());
+    }
+
+    private List<Variant> readIntoList() throws Exception {
+        reader.open(executionContext);
+        List<Variant> variants = new ArrayList<>();
+        Variant variant;
+        while ((variant = reader.read()) != null) {
+            variants.add(variant);
+        }
+        reader.close();
+        return variants;
+    }
+
+    @Test
+    public void linkedSubmittedVariants() throws Exception {
+        Map<String, Variant> variants = readIntoMap();
+        assertEquals(2, variants.size());
+        assertEquals(2, variants.get(RS_1).getSourceEntries().size());
+        assertEquals(1, variants.get(RS_2).getSourceEntries().size());
+    }
+
+    private Map<String, Variant> readIntoMap() throws Exception {
+        reader.open(executionContext);
+        Map<String, Variant> variants = new HashMap<>();
+        Variant variant;
+        while ((variant = reader.read()) != null) {
+            variants.put(variant.getMainId(), variant);
+        }
+        reader.close();
+        return variants;
+    }
+
+    @Test
+    public void queryOtherAssembly() throws Exception {
```

Comment:
I am trying to understand what is unique about this test. Is it to test the results when we query other assemblies besides the default one?

Reply:
Yes, it is only using another assembly besides the default one